be/src/vec/functions/like.cpp

```diff
@@ -129,6 +130,42 @@ Status FunctionLikeBase::vector_vector(const ColumnString::Chars& values,
                                        const ColumnString::Offsets& pattern_offsets,
                                        ColumnUInt8::Container& result, const LikeFn& function,
                                        LikeSearchState* search_state) {
+    // for constant_substring_fn, use long run length search for performance
+    if (constant_substring_fn ==
```

Comment:
Maybe we should move it to a function like named `vector_constant`

Reply:
There are different functions vectorVector, vectorConstant, constantVector for FunctionsStringSearch in clickhouse.

There are some difference between FunctionsStringSearch in clickhouse and like in doris.

We have many constant functions in like, such as constant_equals_fn, constant_starts_with_fn, constant_ends_with_fn, constant_substring_fn. We still need to add  this special branch for constant_substring_fn, since we need to do special optimization for substring.